modelling/src/main/java/org/axonframework/modelling/command/LockingRepository.java

```diff
@@ -125,6 +125,25 @@ protected LockingRepository(Builder<T> builder) {
         }
     }
 
+    @Override
+    protected LockAwareAggregate<T, A> doLoadOrCreate(String aggregateIdentifier, Long expectedVersion,
+                                                      Callable<T> factoryMethod) throws Exception {
+        Lock lock = lockFactory.obtainLock(aggregateIdentifier);
+        try {
+            final A aggregate = doLoadWithLock(aggregateIdentifier, expectedVersion);
+            CurrentUnitOfWork.get().onCleanup(u -> lock.release());
+            return new LockAwareAggregate<>(aggregate, lock);
+        } catch (AggregateNotFoundException ex) {
+            final A aggregate = doCreateNewForLock(factoryMethod);
+            CurrentUnitOfWork.get().onCleanup(u -> lock.release());
+            return new LockAwareAggregate<>(aggregate, lock);
+        } catch (Throwable ex) {
+            logger.debug("Exception occurred while trying to load an aggregate. Releasing lock.", ex);
```

Comment:
I'd assume the debug logging would state something like "while trying to load/add an aggregate".